pkg/output/uncovered.go

```diff
@@ -0,0 +1,441 @@
+	}
+
+	uncoveredInfos := make([]FileUncoveredInfo, 0)
+	for _, profile := range filesToShow {
+		// Check if this profile has any uncovered blocks
+		hasUncovered := false
+		for _, block := range profile.Blocks {
+			if block.Count == 0 {
+				hasUncovered = true
+				break
+			}
+		}
+		
+		if !hasUncovered {
+			continue
+		}
+		
+		info, err := analyzeFileUncovered(profile, cfg)
+		if err != nil {
+			continue // Skip files that can't be read
+		}
+		if len(info.Blocks) > 0 {
+			uncoveredInfos = append(uncoveredInfos, info)
+		}
+	}
+
+	if len(uncoveredInfos) == 0 {
+		fmt.Println("No uncovered lines found!")
+		return nil
+	}
+
+	return displayUncoveredLines(uncoveredInfos, cfg)
+}
+
+// analyzeFileUncovered analyzes a single file to find uncovered lines
+func analyzeFileUncovered(profile *cover.Profile, cfg *config.Config) (FileUncoveredInfo, error) {
+	info := FileUncoveredInfo{
+		FileName: profile.FileName,
+		Blocks:   make([]UncoveredBlock, 0),
+	}
+
+	// Read the source file
+	sourceLines, err := readSourceFile(profile.FileName)
+	if err != nil {
+		return info, err
+	}
+
+	// Create a map to track covered lines
+	coveredLines := make(map[int]bool)
+	for _, block := range profile.Blocks {
+		if block.Count > 0 {
+			for line := block.StartLine; line <= block.EndLine; line++ {
+				coveredLines[line] = true
+			}
+		}
+	}
+
+	// Find uncovered blocks
+	var currentBlock *UncoveredBlock
+	for _, block := range profile.Blocks {
+		if block.Count == 0 {
+			// This block is uncovered
+			for line := block.StartLine; line <= block.EndLine; line++ {
+				if line <= len(sourceLines) {
+					lineContent := sourceLines[line-1] // Convert to 0-based index
+					
+					// Skip lines that shouldn't be considered uncovered
+					if shouldSkipLine(lineContent) {
+						continue
+					}
+					
+					uncoveredLine := UncoveredLine{
+						LineNumber: line,
+						Content:    lineContent,
+						IsCovered:  false,
+					}
+
+					// Group consecutive uncovered lines into blocks
+					if currentBlock == nil || line != currentBlock.EndLine+1 {
+						if currentBlock != nil {
+							// Add context lines to the completed block
+							addContextLines(currentBlock, sourceLines, coveredLines, cfg.UncoveredContext)
+							info.Blocks = append(info.Blocks, *currentBlock)
+						}
+						currentBlock = &UncoveredBlock{
+							StartLine: line,
+							EndLine:   line,
+							Lines:     []UncoveredLine{uncoveredLine},
+						}
+					} else {
+						currentBlock.EndLine = line
+						currentBlock.Lines = append(currentBlock.Lines, uncoveredLine)
+					}
+				}
+			}
+		}
+	}
+
+	if currentBlock != nil {
+		// Add context lines to the final block
+		addContextLines(currentBlock, sourceLines, coveredLines, cfg.UncoveredContext)
+		info.Blocks = append(info.Blocks, *currentBlock)
+	}
+
+	return info, nil
+}
+
+// shouldSkipLine determines if a line should be skipped when showing uncovered lines
+func shouldSkipLine(content string) bool {
```

Comment:
The solution still shows all of these scenarios as uncovered lines when showing uncovered lines... why is that?

Reply:
That fixed it as long as we don't add --uncovered-context lines... When this flag is passed these lines are still shown as uncovered.. There should just be ignore lines ... there should be not color coding or token in the left margin for them when they are displayed due to --uncovered-context